ui/src/app/utils/navigation.tsx

```diff
@@ -34,47 +32,51 @@ export const currentCohortSearchContextStore = new BehaviorSubject<any>(undefine
 export const setSidebarActiveIconStore = new BehaviorSubject<string>(null);
 export const conceptSetUpdating = new BehaviorSubject<boolean>(false);
 
-export const signInStore =
-  new BehaviorSubject<{
-    signOut: Function,
-    profileImage: string,
-  }>({
-    signOut: () => {},
-    profileImage: {} as string,
-  });
+export const useNavigation = () => {
+  const history = useHistory();
 
-/**
- * Slightly stricter variant of Angular's DefaultRouteReuseStrategy. This
- * strategy requires identical params for route reuse, in addition to route
- * config matching.
- *
- * This avoid bugs from invalid parameter assumptions in views and for the most
- * part should not affect performance, based on parameter usage in the
- * application.
- */
-export class WorkbenchRouteReuseStrategy extends RouteReuseStrategy {
-  // The following methods are copied from Angular's DefaultRouteReuseStrategy.
-  shouldDetach(route: ActivatedRouteSnapshot): boolean { return false; }
-  store(route: ActivatedRouteSnapshot, detachedTree: DetachedRouteHandle): void {}
-  shouldAttach(route: ActivatedRouteSnapshot): boolean { return false; }
-  retrieve(route: ActivatedRouteSnapshot): DetachedRouteHandle|null { return null; }
-
-  shouldReuseRoute(future: ActivatedRouteSnapshot, curr: ActivatedRouteSnapshot): boolean {
-    return future.routeConfig === curr.routeConfig
-      && (fp.isEqual(future.params, curr.params) || curr.data.shouldReuse);
-  }
-}
+  const navigateByUrl = (url, extras?: NavigateExtras) => {
+    url = '/' + url.replace(/^\//, '');
```

Comment:
Per my other comments, looks like the leading / is just ignored here. Probably better to make all the `navigate()`s consistent then